cmd/ketch/job_list.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/shipa-corp/ketch/cmd/ketch/output"
+	ketchv1 "github.com/shipa-corp/ketch/internal/api/v1beta1"
+)
+
+const jobListHelp = `
+List all jobs.
+`
+
+type jobListOutput struct {
+	Name        string `json:"name"`
+	Version     string `json:"version"`
+	Framework   string `json:"framework"`
+	Description string `json:"description"`
+}
+
+func newJobListCmd(cfg config, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all jobs.",
+		Long:  jobListHelp,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return jobList(cmd.Context(), cfg, out)
+		},
+	}
+	return cmd
+}
+
+func jobList(ctx context.Context, cfg config, out io.Writer) error {
+	jobs := ketchv1.JobList{}
+	if err := cfg.Client().List(ctx, &jobs); err != nil {
+		return fmt.Errorf("failed to get list of jobs: %w", err)
+	}
+	return output.Write(generateJobListOutput(jobs), out, "column")
+}
+
+func generateJobListOutput(jobs ketchv1.JobList) []jobListOutput {
+	var output []jobListOutput
+	for _, item := range jobs.Items {
+		output = append(output, jobListOutput{
+			Name:        item.Name,
+			Version:     item.Spec.Version,
+			Framework:   item.Spec.Framework,
+			Description: item.Spec.Description,
+		})
+	}
+	return output
+}
+
+func jobListNames(cfg config, nameFilter ...string) ([]string, error) {
+	jobs := ketchv1.JobList{}
+	if err := cfg.Client().List(context.TODO(), &jobs); err != nil {
+		return nil, fmt.Errorf("failed to list apps: %w", err)
```

Comment:
should it be `failed to list jobs`?

Reply:
Good catch. It's possible I was copy/pasting.